### DISCUSSION TITLE
Building with Android error java.security.InvalidAlgorithmParameterException: the trustAnchors parameter must be non-empty

### DISCUSSION THREAD

**Discussion by BoosterUP1256** [HaxeFlixel/flixel#2348] [Q&A] [answered]
I am trying to build my project for android but I keep getting this error:

FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring root project 'bin'.
> Could not resolve all artifacts for configuration ':classpath'.
   > Could not resolve com.android.tools.build:gradle:3.5.1.
     Required by:
         project :
      > Could not resolve com.android.tools.build:gradle:3.5.1.
         > Could not get resource 'https://jcenter.bintray.com/com/android/tools/build/gradle/3.5.1/gradle-3.5.1.pom'.
            > Could not GET 'https://jcenter.bintray.com/com/android/tools/build/gradle/3.5.1/gradle-3.5.1.pom'.
               > java.lang.RuntimeException: Unexpected error: java.security.InvalidAlgorithmParameterException: the trustAnchors parameter must be non-empty
      > Could not resolve com.android.tools.build:gradle:3.5.1.
         > Could not get resource 'https://dl.google.com/dl/android/maven2/com/android/tools/build/gradle/3.5.1/gradle-3.5.1.pom'.
            > Could not GET 'https://dl.google.com/dl/android/maven2/com/android/tools/build/gradle/3.5.1/gradle-3.5.1.pom'.
               > java.lang.RuntimeException: Unexpected error: java.security.InvalidAlgorithmParameterException: the trustAnchors parameter must be non-empty

I am using openJDK8. My android sdk manager setup looks like this.
![Screenshot 2021-07-18 134837](https://user-images.githubusercontent.com/84538912/126077202-09bf4ea8-5c2c-49f9-9203-3d369f31445b.png)

![image](https://user-images.githubusercontent.com/84538912/126077214-c4573425-373f-4c54-a761-2425eac50ac2.png)

I don't have any previous experience developing for android so sorry if this is a simple problem.

**Comment by DinoMah** [comment, marked as answer]
Hello, um I just followed this tutorial -> [https://lime.software/docs/advanced-setup/android/](tutorial) and had no trouble building, note that lime requires _**NDK**_ version _**r15c**_, android studio doesn't offer the r15c version.
If you don't know how to setup ndk manually, just ask and I'll help you.

**Reply by Gama11** [reply]
You don't need an NDK that ancient, r21e should also work: https://github.com/HaxeFoundation/hxcpp/issues/706#issuecomment-856144152

**Reply by BoosterUP1256** [reply]
Sorry, I forgot to specify in my original question. I am already using NDK version r15c. Here is my lime setup.
![image](https://user-images.githubusercontent.com/84538912/126340735-e0c18e55-c8d1-4ece-9447-c582d5a4e949.png)
as far as setting up my NDK, the only thing I did was set the path to it in limes setup. Is there something else I have to do?

**Reply by DinoMah** [reply]
> 
> 
> You don't need an NDK that ancient, r21e should also work: [HaxeFoundation/hxcpp#706 (comment)](https://github.com/HaxeFoundation/hxcpp/issues/706#issuecomment-856144152)

Didn't know @Gama11, they need to update info regularly 😕, also @BoosterUP1256 I'm noticing you have your jdk in your downloads folder, why? 🤔😅, I have my NDK in the NDK folder of Android SDK if you find it useful

**Reply by BoosterUP1256** [reply]
I don't really work with java that much and when I do I have always used the latest version of the JDK. The lime documentation said it was best to work with java 8 to build for android. I did not want to create an account with oracle to download older versions of java so I downloaded the openJDK. I guess I just never bothered putting it in my java folder because I just downloaded it for building with Haxeflixel. Although, your right. I probably should put it in a better location. Anyway I tried building with NDK version r15c and r21e and got the same error I did before  both times. I have also tried using the latest android API as well as versions 26 and version 19 with the same result I am really not sure what is going on?

**Reply by DinoMah** [reply]
Oh, didn't notice it was OpenJDK, I'm using the AdoptOpenJDK one, don't know if there's a difference between them but the installer sets the JAVA_HOME and the path variable.
Also, I tried later versions of the NDK but none of them worked so stick around r15c if you don't want any trouble.

**Reply by BoosterUP1256** [reply]
so I switched from the OpenJDK binarys listed on Oracles website to AdoptOpenJDK and everything built fine. I still have to get an emulator up and running but everything seems to have compiled fine. I think it has something to do with setting JAVA_HOME in my environment variables (which the AdoptOpenJDK installer sets for you)
